ZAPD/ZActorList.cpp

```diff
@@ -0,0 +1,194 @@
+			WarningType::InvalidAttributeValue, parent, this, rawDataIndex,
+			StringHelper::Sprintf("invalid value '%d' found for 'NumPaths' attribute", numActors),
+			"Should be at least '1'");
+	}
+}
+
+void ZActorList::ParseRawData()
+{
+	ZResource::ParseRawData();
+
+	offset_t currentPtr = rawDataIndex;
+	size_t largestlength = 0;
+
+	for (size_t i = 0; i < numActors; i++)
+	{
+		ActorSpawnEntry entry(parent->GetRawData(), currentPtr);
+
+		currentPtr += entry.GetRawDataSize();
+		actors.push_back(entry);
+
+		size_t actorNameLength = ZNames::GetActorName(entry.GetActorId()).size();
+		if (actorNameLength > largestlength)
+			largestlength = actorNameLength;
+	}
+
+	for (auto& entry : actors)
+	{
+		entry.SetLargestActorName(largestlength);
+	}
+}
+
+Declaration* ZActorList::DeclareVar(const std::string& prefix, const std::string& bodyStr)
+{
+	std::string auxName = name;
+
+	if (name == "")
+		auxName = GetDefaultName(prefix);
+
+	Declaration* decl =
+		parent->AddDeclarationArray(rawDataIndex, GetDeclarationAlignment(), GetRawDataSize(),
+	                                GetSourceTypeName(), name, GetActorListArraySize(), bodyStr);
+	decl->staticConf = staticConf;
+
+	return decl;
+}
+
+std::string ZActorList::GetBodySourceCode() const
+{
+	std::string declaration;
+
+	size_t index = 0;
+	for (auto& entry : actors)
+	{
+		declaration += StringHelper::Sprintf("\t{ %s },", entry.GetBodySourceCode().c_str());
+
+		if (index < actors.size() - 1)
+			declaration += "\n";
+
+		index++;
+	}
+
+	return declaration;
+}
+
+std::string ZActorList::GetSourceTypeName() const
+{
+	return actors.front().GetSourceTypeName();
+}
+
+ZResourceType ZActorList::GetResourceType() const
+{
+	return ZResourceType::ActorList;
+}
+
+size_t ZActorList::GetRawDataSize() const
+{
+	return actors.size() * actors.front().GetRawDataSize();
+}
+
+size_t ZActorList::GetActorListArraySize() const
+{
+	size_t actorCount = 0;
+
+	// Doing an else-if here so we only do the loop when the game is SW97.
+	// Actor 0x22 is removed from SW97, so we need to ensure that we don't increment the actor count
+	// for it.
+	if (Globals::Instance->game == ZGame::OOT_SW97)
+	{
+		actorCount = 0;
+
+		for (const auto& entry : actors)
+			if (entry.GetActorId() != 0x22)
+				actorCount++;
+	}
+	else
+	{
+		actorCount = actors.size();
+	}
+
+	return actorCount;
+}
+
+/* ActorSpawnEntry */
+
+ActorSpawnEntry::ActorSpawnEntry(const std::vector<uint8_t>& rawData, uint32_t rawDataIndex)
+{
+	actorNum = BitConverter::ToInt16BE(rawData, rawDataIndex + 0);
+	posX = BitConverter::ToInt16BE(rawData, rawDataIndex + 2);
+	posY = BitConverter::ToInt16BE(rawData, rawDataIndex + 4);
+	posZ = BitConverter::ToInt16BE(rawData, rawDataIndex + 6);
+	rotX = BitConverter::ToUInt16BE(rawData, rawDataIndex + 8);
+	rotY = BitConverter::ToUInt16BE(rawData, rawDataIndex + 10);
+	rotZ = BitConverter::ToUInt16BE(rawData, rawDataIndex + 12);
+	initVar = BitConverter::ToInt16BE(rawData, rawDataIndex + 14);
```

Comment:
Since this is called `params` in OOT it should probably be that here. Same with `actorNum I think it should be `actorId`

Reply:
Just this and then its good to me.